app/mapping/AssetExtractor.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mapping
+
+import mapping.PlaybackExtractionErrors.{FailedToExtractData, PlaybackExtractionError}
+import mapping.PlaybackImplicits.AddressConverter
+import models.UserAnswers
+import models.http.{DisplayNonEEABusinessType, DisplayTrustAssets}
+import pages.assets.nonEeaBusiness._
+import play.api.Logging
+
+import scala.util.{Failure, Success, Try}
+
+class AssetExtractor extends Logging {
+
+  def extract(answers: UserAnswers, data: Option[DisplayTrustAssets]): Either[PlaybackExtractionError, UserAnswers] = {
+    if (data.isDefined) {
+      val updated = data.get.nonEEABusiness.zipWithIndex.foldLeft[Try[UserAnswers]](Success(answers)) {
+        case (answers, (entity, index)) =>
+          updateUserAnswers(answers, entity, index)
+      }
+
+      updated match {
+        case Success(a) =>
+          Right(a)
+        case Failure(exception) =>
+          logger.warn(s"[UTR/URN: ${answers.identifier}] failed to extract data due to ${exception.getMessage}")
+          Left(FailedToExtractData(DisplayNonEEABusinessType.toString))
+      }
+    } else {
+      Right(answers)
+    }
+  }
```

Comment:
I believe if you make AssetsExtractor extend `PlaybackExtractor[DisplayNonEEABusinessType]` then you can get rid of this entire extract method, stick an override on the def updateUserAnswers, add `override val optionalEntity: Boolean = true` and change the UserAnswersExtractor to `data.trust.assets.map(_.nonEEABusiness).getOrElse(Nil)`. See the `OtherIndividualExtractor` for reference. If it doesn't work then no worries